### ISSUE TITLE
Overrides.json is not respected by the keyboard shortcuts extensions

### ISSUE THREAD

**Issue by divyansshhh** [jupyterlab/jupyterlab#15713]
## Description

I have the following snippet in my overrides.json but it seems to get ignored by the shortcuts-extension.

```
{
  "@jupyterlab/shortcuts-extension:shortcuts": {
    shortcuts: [
      {
        "args": {},
        "command": "documentsearch:start",
        "keys": [
            "Accel F"
        ],
        "disabled": true,
        "selector": ".jp-mod-searchable"
      },
    ],
  },
}
```

## Reproduce

1. Copy the above snippet into `<sys-prefix>/jupyter/lab/settings/overrides.json5`.
2. Launch jupyterlab.
3. Open a notebook.
4. Use the Ctrl F shortcut.

## Expected behavior
I would expect the keyboard shortcut to launch the browser's find instead of the one provided by jupyterlab

## Context
- Jupyterlab version: 4.1.0rc0

**Comment by krassowski** [comment]
I can reproduce this in 4.0.rc0 and 4.0.11 but not in 3.6.x. This is not limited to disabling shortcuts (as in the example above) but also affects changing shortcuts in general. The overrides for other plugins are respected.

**Comment by krassowski** [comment]
Notes:
- the ability for `overrides.json` was introduced in https://github.com/jupyterlab/jupyterlab/pull/9858 (v3.1.0)
- the overrides content is properly sent form backend
- it appears to happen prior to `@jupyterlab/shortcuts-extension:shortcuts` activation at which point the default used to be already changed in 3.6.x but is not in 4.0.x, suggesting that the issue is in transformation step.
- it is NOT a regression from https://github.com/jupyterlab/jupyterlab/pull/14195 since it was included only in [v4.0.0rc0](https://github.com/jupyterlab/jupyterlab/releases/tag/v4.0.0rc0) but [v4.0.0b2](https://github.com/jupyterlab/jupyterlab/releases/tag/v4.0.0b2) is not affected
- from bisection, this was introduced in [v4.0.0b1](https://github.com/jupyterlab/jupyterlab/releases/tag/v4.0.0b1) (https://github.com/jupyterlab/jupyterlab/compare/v4.0.0b0...v4.0.0b1) which included PRs modifying how settings are fetched from the server and processed:
   - https://github.com/jupyterlab/jupyterlab/pull/14211
   - https://github.com/jupyterlab/jupyterlab/pull/14195

The first time `reconcileShortcuts` gets called the defaults from overrides are there. It then gets called again and again - dozens of times - on the page load and in the second and consecutive calls the defaults are missing the overrides.

Initial load:

![Screenshot from 2024-01-30 14-11-32](https://github.com/jupyterlab/jupyterlab/assets/5832902/e8b2d621-6c61-423b-ba64-7eabeda7e6b3)

Example subsequent load:

![Screenshot from 2024-01-30 14-12-17](https://github.com/jupyterlab/jupyterlab/assets/5832902/3a60324a-e4b6-4f0a-a4f8-d029a5aa8835)

The subsequent loads happen when settings of any other plugin are fetched. This is expected because each plugin can contribute shortcuts. 

To fix this issue we need to slightly improve what PR https://github.com/jupyterlab/jupyterlab/pull/14195 attempted to do in `shortcuts-extension/src/index.ts` file. I will open a PR.
